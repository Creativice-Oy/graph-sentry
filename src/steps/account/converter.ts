import {
  createIntegrationEntity,
  Entity,
} from '@jupiterone/integration-sdk-core';
import { SentryOrganization } from '../../types';

import { Entities } from '../constants';

export function createOrganizationEntity(
  organization: SentryOrganization,
): Entity {
  return createIntegrationEntity({
    entityData: {
      source: organization,
      assign: {
        _key: `sentry-organization-${organization.id}`,
        _type: Entities.ORGANIZATION._type,
        _class: Entities.ORGANIZATION._class,
        slug: organization.slug,
      },
    },
  });
}
