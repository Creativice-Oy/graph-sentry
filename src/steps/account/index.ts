import {
  IntegrationStep,
  IntegrationStepExecutionContext,
} from '@jupiterone/integration-sdk-core';

import { createAPIClient } from '../../client';
import { IntegrationConfig } from '../../config';
import { Entities, Steps } from '../constants';
import { createOrganizationEntity } from './converter';

export async function fetchOrganizations({
  instance,
  jobState,
  logger,
}: IntegrationStepExecutionContext<IntegrationConfig>) {
  const apiClient = createAPIClient(instance.config, logger);

  // const orgData = (await jobState.getData(ACCOUNT_ENTITY_KEY)) as Entity;

  await apiClient.iterateOrganizations(async (orgData) => {
    //TODO, should we be filtering by a listed organization ID in the config?
    await jobState.addEntity(createOrganizationEntity(orgData));
  });
}

export const organizationSteps: IntegrationStep<IntegrationConfig>[] = [
  {
    id: Steps.ORGANIZATIONS,
    name: 'Fetch Organization Details',
    entities: [Entities.ORGANIZATION],
    relationships: [],
    dependsOn: [],
    executionHandler: fetchOrganizations,
  },
];
