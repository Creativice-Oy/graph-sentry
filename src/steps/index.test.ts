import { createMockStepExecutionContext } from '@jupiterone/integration-sdk-testing';

import { IntegrationConfig } from '../config';
import { fetchOrganizations } from './account';
import {
  fetchUsers,
  fetchTeams,
  fetchUserAssignments,
  fetchProjects,
} from './access';
import { integrationConfig } from '../../test/config';
import { Recording, setupSentryRecording } from '../../test/recording';

let recording: Recording;

test('should collect data', async () => {
  const context = createMockStepExecutionContext<IntegrationConfig>({
    instanceConfig: integrationConfig,
  });
  recording = setupSentryRecording({
    directory: __dirname,
    name: 'should collect data',
  });

  // Simulates dependency graph execution.
  // See https://github.com/JupiterOne/sdk/issues/262.
  await fetchOrganizations(context);
  await fetchProjects(context);
  await fetchTeams(context);
  await fetchUsers(context);
  await fetchUserAssignments(context);

  // Review snapshot, failure is a regression
  expect({
    numCollectedEntities: context.jobState.collectedEntities.length,
    numCollectedRelationships: context.jobState.collectedRelationships.length,
    collectedEntities: context.jobState.collectedEntities,
    collectedRelationships: context.jobState.collectedRelationships,
    encounteredTypes: context.jobState.encounteredTypes,
  }).toMatchSnapshot();

  const accounts = context.jobState.collectedEntities.filter((e) =>
    e._class.includes('Account'),
  );
  expect(accounts.length).toBeGreaterThan(0);
  expect(accounts).toMatchGraphObjectSchema({
    _class: ['Account'],
    schema: {
      additionalProperties: false,
      properties: {
        _type: { const: 'sentry_organization' },
        slug: { type: 'string' },
        _rawData: {
          type: 'array',
          items: { type: 'object' },
        },
      },
      required: ['slug'],
    },
  });

  const users = context.jobState.collectedEntities.filter((e) =>
    e._class.includes('User'),
  );
  expect(users.length).toBeGreaterThan(0);
  expect(users).toMatchGraphObjectSchema({
    _class: ['User'],
    schema: {
      additionalProperties: false,
      properties: {
        _type: { const: 'sentry_member' },
        username: { type: 'string' },
        _rawData: {
          type: 'array',
          items: { type: 'object' },
        },
      },
      required: ['username'],
    },
  });

  const userGroups = context.jobState.collectedEntities.filter((e) =>
    e._class.includes('UserGroup'),
  );
  expect(userGroups.length).toBeGreaterThan(0);
  expect(userGroups).toMatchGraphObjectSchema({
    _class: ['UserGroup'],
    schema: {
      additionalProperties: false,
      properties: {
        _type: { const: 'sentry_team' },
        slug: { type: 'string' },
        _rawData: {
          type: 'array',
          items: { type: 'object' },
        },
      },
      required: ['slug'],
    },
  });

  const projects = context.jobState.collectedEntities.filter((e) =>
    e._class.includes('Project'),
  );
  expect(projects.length).toBeGreaterThan(0);
  expect(projects).toMatchGraphObjectSchema({
    _class: ['Project'],
    schema: {
      additionalProperties: false,
      properties: {
        _type: { const: 'sentry_project' },
        name: { type: 'string' },
        _rawData: {
          type: 'array',
          items: { type: 'object' },
        },
      },
      required: ['name'],
    },
  });

  if (recording) {
    await recording.stop();
  }
});
