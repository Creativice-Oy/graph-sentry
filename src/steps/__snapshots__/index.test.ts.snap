// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Account",
      ],
      "_key": "sentry-organization-1052715",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "access": Array [
              "project:read",
              "team:read",
              "org:read",
              "member:read",
            ],
            "alertsMemberWrite": true,
            "allowJoinRequests": true,
            "allowSharedIssues": true,
            "apdexThreshold": 300,
            "attachmentsRole": "member",
            "availableRoles": Array [
              Object {
                "id": "billing",
                "name": "Billing",
              },
              Object {
                "id": "member",
                "name": "Member",
              },
              Object {
                "id": "admin",
                "name": "Admin",
              },
              Object {
                "id": "manager",
                "name": "Manager",
              },
              Object {
                "id": "owner",
                "name": "Owner",
              },
            ],
            "avatar": Object {
              "avatarType": "letter_avatar",
              "avatarUuid": null,
            },
            "dataScrubber": false,
            "dataScrubberDefaults": false,
            "dateCreated": "2021-10-26T20:55:38.686124Z",
            "debugFilesRole": "admin",
            "defaultRole": "member",
            "enhancedPrivacy": false,
            "eventsMemberAdmin": true,
            "experiments": Object {
              "TargetedUpsellModalExperiment": 1,
              "TrialLabelExperiment": 0,
            },
            "features": Array [
              "performance-tag-page",
              "discover-top-events",
              "shared-issues",
              "performance-mobile-vitals",
              "unified-span-view",
              "open-membership",
              "minute-resolution-sessions",
              "event-attachments-viewer",
              "semver",
              "custom-event-title",
              "event-attachments",
              "release-adoption-chart",
              "advanced-search",
              "performance-view",
              "chart-unfurls",
              "performance-events-page",
              "org-subdomains",
              "issue-percent-filters",
              "invite-members-rate-limits",
              "release-adoption-stage",
              "images-loaded-v2",
              "release-comparison",
              "performance-ops-breakdown",
              "onboarding",
              "mobile-app",
              "symbol-sources",
              "crash-rate-alerts",
              "sso-scim",
              "metrics-extraction",
              "integrations-stacktrace-link",
              "connect-discover-and-dashboards",
              "unhandled-issue-flag",
            ],
            "id": "1052715",
            "isDefault": false,
            "isEarlyAdopter": false,
            "name": "JupiterOne Integration Development",
            "onboardingTasks": Array [
              Object {
                "completionSeen": null,
                "data": Object {},
                "dateCompleted": "2021-10-27T11:58:01.475644Z",
                "status": "pending",
                "task": "setup_second_platform",
                "user": Object {
                  "avatar": Object {
                    "avatarType": "letter_avatar",
                    "avatarUuid": null,
                  },
                  "avatarUrl": "https://secure.gravatar.com/avatar/5abd8230461366dd40ab803831f7ad97?s=32&d=mm",
                  "dateJoined": "2021-10-26T21:08:09.627990Z",
                  "email": "adam.pierson@jupiterone.com",
                  "emails": Array [
                    Object {
                      "email": "adam.pierson@jupiterone.com",
                      "id": "1735481",
                      "is_verified": true,
                    },
                  ],
                  "experiments": Object {},
                  "has2fa": true,
                  "hasPasswordAuth": true,
                  "id": "1601268",
                  "isActive": true,
                  "isManaged": false,
                  "isStaff": false,
                  "isSuperuser": false,
                  "lastActive": "2021-11-09T19:13:38.496885Z",
                  "lastLogin": "2021-11-02T17:49:11.974792Z",
                  "name": "Adam Pierson",
                  "username": "adam.pierson@jupiterone.com",
                },
              },
              Object {
                "completionSeen": null,
                "data": Object {
                  "invited_member_id": 10763262,
                },
                "dateCompleted": "2021-10-26T21:08:10.222422Z",
                "status": "complete",
                "task": "invite_member",
                "user": Object {
                  "avatar": Object {
                    "avatarType": "letter_avatar",
                    "avatarUuid": null,
                  },
                  "avatarUrl": "https://secure.gravatar.com/avatar/6c6d286122b52413957d2e2ed73deefc?s=32&d=mm",
                  "dateJoined": "2021-10-26T20:55:38.564265Z",
                  "email": "austin.kelleher@jupiterone.dev",
                  "emails": Array [
                    Object {
                      "email": "austin.kelleher@jupiterone.dev",
                      "id": "1735468",
                      "is_verified": true,
                    },
                  ],
                  "experiments": Object {},
                  "has2fa": false,
                  "hasPasswordAuth": true,
                  "id": "1601255",
                  "isActive": true,
                  "isManaged": false,
                  "isStaff": false,
                  "isSuperuser": false,
                  "lastActive": "2021-11-09T18:18:43.196636Z",
                  "lastLogin": "2021-11-08T19:34:22.952705Z",
                  "name": "Austin Kelleher",
                  "username": "austin.kelleher@jupiterone.dev",
                },
              },
              Object {
                "completionSeen": null,
                "data": Object {
                  "platform": "node",
                },
                "dateCompleted": "2021-10-26T20:59:35.888000Z",
                "status": "complete",
                "task": "send_first_event",
                "user": null,
              },
              Object {
                "completionSeen": null,
                "data": Object {},
                "dateCompleted": "2021-10-26T20:59:35.840192Z",
                "status": "complete",
                "task": "setup_transactions",
                "user": null,
              },
              Object {
                "completionSeen": null,
                "data": Object {},
                "dateCompleted": "2021-10-26T20:56:06.175633Z",
                "status": "complete",
                "task": "create_project",
                "user": Object {
                  "avatar": Object {
                    "avatarType": "letter_avatar",
                    "avatarUuid": null,
                  },
                  "avatarUrl": "https://secure.gravatar.com/avatar/6c6d286122b52413957d2e2ed73deefc?s=32&d=mm",
                  "dateJoined": "2021-10-26T20:55:38.564265Z",
                  "email": "austin.kelleher@jupiterone.dev",
                  "emails": Array [
                    Object {
                      "email": "austin.kelleher@jupiterone.dev",
                      "id": "1735468",
                      "is_verified": true,
                    },
                  ],
                  "experiments": Object {},
                  "has2fa": false,
                  "hasPasswordAuth": true,
                  "id": "1601255",
                  "isActive": true,
                  "isManaged": false,
                  "isStaff": false,
                  "isSuperuser": false,
                  "lastActive": "2021-11-09T18:18:43.196636Z",
                  "lastLogin": "2021-11-08T19:34:22.952705Z",
                  "name": "Austin Kelleher",
                  "username": "austin.kelleher@jupiterone.dev",
                },
              },
            ],
            "openMembership": true,
            "pendingAccessRequests": 0,
            "projects": Array [
              Object {
                "dateCreated": "2021-10-26T20:56:06.026232Z",
                "environments": Array [
                  "production",
                ],
                "eventProcessing": Object {
                  "symbolicationDegraded": false,
                },
                "features": Array [
                  "alert-filters",
                  "issue-alerts-targeting",
                  "minidump",
                  "race-free-group-creation",
                  "similarity-indexing",
                  "similarity-view",
                ],
                "firstEvent": "2021-10-26T20:59:35.888000Z",
                "firstTransactionEvent": true,
                "hasAccess": true,
                "hasSessions": false,
                "hasUserReports": false,
                "id": "6036546",
                "isBookmarked": false,
                "isMember": false,
                "latestDeploys": null,
                "latestRelease": null,
                "name": "jupiterone-integration-develop",
                "platform": "node",
                "platforms": Array [
                  "node",
                ],
                "slug": "jupiterone-integration-develop",
                "team": Object {
                  "id": "1252019",
                  "name": "JupiterOne Integration Development",
                  "slug": "jupiterone-integration-develop",
                },
                "teams": Array [
                  Object {
                    "id": "1252019",
                    "name": "JupiterOne Integration Development",
                    "slug": "jupiterone-integration-develop",
                  },
                ],
              },
              Object {
                "dateCreated": "2021-11-01T21:12:05.790523Z",
                "environments": Array [],
                "eventProcessing": Object {
                  "symbolicationDegraded": false,
                },
                "features": Array [
                  "alert-filters",
                  "issue-alerts-targeting",
                  "minidump",
                  "race-free-group-creation",
                  "similarity-indexing",
                  "similarity-view",
                ],
                "firstEvent": null,
                "firstTransactionEvent": false,
                "hasAccess": true,
                "hasSessions": false,
                "hasUserReports": false,
                "id": "6044646",
                "isBookmarked": false,
                "isMember": false,
                "latestDeploys": null,
                "latestRelease": null,
                "name": "testproject2",
                "platform": "javascript",
                "platforms": Array [],
                "slug": "testproject2",
                "team": Object {
                  "id": "1254438",
                  "name": "jupiterone-dev-testteam1",
                  "slug": "jupiterone-dev-testteam1",
                },
                "teams": Array [
                  Object {
                    "id": "1254438",
                    "name": "jupiterone-dev-testteam1",
                    "slug": "jupiterone-dev-testteam1",
                  },
                ],
              },
              Object {
                "dateCreated": "2021-10-27T11:58:01.388720Z",
                "environments": Array [],
                "eventProcessing": Object {
                  "symbolicationDegraded": false,
                },
                "features": Array [
                  "alert-filters",
                  "issue-alerts-targeting",
                  "minidump",
                  "race-free-group-creation",
                  "similarity-indexing",
                  "similarity-view",
                ],
                "firstEvent": null,
                "firstTransactionEvent": false,
                "hasAccess": true,
                "hasSessions": false,
                "hasUserReports": false,
                "id": "6037645",
                "isBookmarked": false,
                "isMember": false,
                "latestDeploys": null,
                "latestRelease": null,
                "name": "testpythonproject1",
                "platform": "python",
                "platforms": Array [],
                "slug": "testpythonproject1",
                "team": Object {
                  "id": "1252019",
                  "name": "JupiterOne Integration Development",
                  "slug": "jupiterone-integration-develop",
                },
                "teams": Array [
                  Object {
                    "id": "1252019",
                    "name": "JupiterOne Integration Development",
                    "slug": "jupiterone-integration-develop",
                  },
                ],
              },
            ],
            "quota": Object {
              "accountLimit": 0,
              "maxRate": null,
              "maxRateInterval": 60,
              "projectLimit": 100,
            },
            "relayPiiConfig": null,
            "require2FA": false,
            "requireEmailVerification": false,
            "safeFields": Array [],
            "scrapeJavaScript": true,
            "scrubIPAddresses": false,
            "sensitiveFields": Array [],
            "slug": "jupiterone-integration-develop",
            "status": Object {
              "id": "active",
              "name": "active",
            },
            "storeCrashReports": 0,
            "teams": Array [
              Object {
                "avatar": Object {
                  "avatarType": "letter_avatar",
                  "avatarUuid": null,
                },
                "dateCreated": "2021-10-28T16:21:51.455854Z",
                "hasAccess": true,
                "id": "1254438",
                "isMember": false,
                "isPending": false,
                "memberCount": 1,
                "name": "jupiterone-dev-testteam1",
                "slug": "jupiterone-dev-testteam1",
              },
              Object {
                "avatar": Object {
                  "avatarType": "letter_avatar",
                  "avatarUuid": null,
                },
                "dateCreated": "2021-10-26T20:55:38.704349Z",
                "hasAccess": true,
                "id": "1252019",
                "isMember": false,
                "isPending": false,
                "memberCount": 2,
                "name": "JupiterOne Integration Development",
                "slug": "jupiterone-integration-develop",
              },
            ],
            "trustedRelays": Array [],
          },
        },
      ],
      "_type": "sentry_organization",
      "active": undefined,
      "createdOn": undefined,
      "displayName": "JupiterOne Integration Development",
      "id": "1052715",
      "name": "JupiterOne Integration Development",
      "slug": "jupiterone-integration-develop",
    },
    Object {
      "_class": Array [
        "Project",
      ],
      "_key": "sentry-project-6036546",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "dateCreated": "2021-10-26T20:56:06.026232Z",
            "environments": Array [
              "production",
            ],
            "eventProcessing": Object {
              "symbolicationDegraded": false,
            },
            "features": Array [
              "alert-filters",
              "issue-alerts-targeting",
              "minidump",
              "race-free-group-creation",
              "similarity-indexing",
              "similarity-view",
            ],
            "firstEvent": "2021-10-26T20:59:35.888000Z",
            "firstTransactionEvent": true,
            "hasAccess": true,
            "hasSessions": false,
            "hasUserReports": false,
            "id": "6036546",
            "isBookmarked": false,
            "isMember": false,
            "latestDeploys": null,
            "latestRelease": null,
            "name": "jupiterone-integration-develop",
            "platform": "node",
            "platforms": Array [
              "node",
            ],
            "slug": "jupiterone-integration-develop",
            "team": Object {
              "id": "1252019",
              "name": "JupiterOne Integration Development",
              "slug": "jupiterone-integration-develop",
            },
            "teams": Array [
              Object {
                "id": "1252019",
                "name": "JupiterOne Integration Development",
                "slug": "jupiterone-integration-develop",
              },
            ],
          },
        },
      ],
      "_type": "sentry_project",
      "createdOn": undefined,
      "displayName": "jupiterone-integration-develop",
      "id": "6036546",
      "name": "jupiterone-integration-develop",
    },
    Object {
      "_class": Array [
        "Project",
      ],
      "_key": "sentry-project-6044646",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "dateCreated": "2021-11-01T21:12:05.790523Z",
            "environments": Array [],
            "eventProcessing": Object {
              "symbolicationDegraded": false,
            },
            "features": Array [
              "alert-filters",
              "issue-alerts-targeting",
              "minidump",
              "race-free-group-creation",
              "similarity-indexing",
              "similarity-view",
            ],
            "firstEvent": null,
            "firstTransactionEvent": false,
            "hasAccess": true,
            "hasSessions": false,
            "hasUserReports": false,
            "id": "6044646",
            "isBookmarked": false,
            "isMember": false,
            "latestDeploys": null,
            "latestRelease": null,
            "name": "testproject2",
            "platform": "javascript",
            "platforms": Array [],
            "slug": "testproject2",
            "team": Object {
              "id": "1254438",
              "name": "jupiterone-dev-testteam1",
              "slug": "jupiterone-dev-testteam1",
            },
            "teams": Array [
              Object {
                "id": "1254438",
                "name": "jupiterone-dev-testteam1",
                "slug": "jupiterone-dev-testteam1",
              },
            ],
          },
        },
      ],
      "_type": "sentry_project",
      "createdOn": undefined,
      "displayName": "testproject2",
      "id": "6044646",
      "name": "testproject2",
    },
    Object {
      "_class": Array [
        "Project",
      ],
      "_key": "sentry-project-6037645",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "dateCreated": "2021-10-27T11:58:01.388720Z",
            "environments": Array [],
            "eventProcessing": Object {
              "symbolicationDegraded": false,
            },
            "features": Array [
              "alert-filters",
              "issue-alerts-targeting",
              "minidump",
              "race-free-group-creation",
              "similarity-indexing",
              "similarity-view",
            ],
            "firstEvent": null,
            "firstTransactionEvent": false,
            "hasAccess": true,
            "hasSessions": false,
            "hasUserReports": false,
            "id": "6037645",
            "isBookmarked": false,
            "isMember": false,
            "latestDeploys": null,
            "latestRelease": null,
            "name": "testpythonproject1",
            "platform": "python",
            "platforms": Array [],
            "slug": "testpythonproject1",
            "team": Object {
              "id": "1252019",
              "name": "JupiterOne Integration Development",
              "slug": "jupiterone-integration-develop",
            },
            "teams": Array [
              Object {
                "id": "1252019",
                "name": "JupiterOne Integration Development",
                "slug": "jupiterone-integration-develop",
              },
            ],
          },
        },
      ],
      "_type": "sentry_project",
      "createdOn": undefined,
      "displayName": "testpythonproject1",
      "id": "6037645",
      "name": "testpythonproject1",
    },
    Object {
      "_class": Array [
        "UserGroup",
      ],
      "_key": "sentry-team-1254438",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "avatar": Object {
              "avatarType": "letter_avatar",
              "avatarUuid": null,
            },
            "dateCreated": "2021-10-28T16:21:51.455854Z",
            "externalTeams": Array [],
            "hasAccess": true,
            "id": "1254438",
            "isMember": false,
            "isPending": false,
            "memberCount": 1,
            "name": "jupiterone-dev-testteam1",
            "projects": Array [
              Object {
                "avatar": Object {
                  "avatarType": "letter_avatar",
                  "avatarUuid": null,
                },
                "color": "#bf533f",
                "dateCreated": "2021-11-01T21:12:05.790523Z",
                "features": Array [
                  "alert-filters",
                  "issue-alerts-targeting",
                  "minidump",
                  "race-free-group-creation",
                  "similarity-indexing",
                  "similarity-view",
                ],
                "firstEvent": null,
                "firstTransactionEvent": false,
                "hasAccess": true,
                "hasSessions": false,
                "id": "6044646",
                "isBookmarked": false,
                "isInternal": false,
                "isMember": false,
                "isPublic": false,
                "name": "testproject2",
                "platform": "javascript",
                "slug": "testproject2",
                "status": "active",
              },
            ],
            "slug": "jupiterone-dev-testteam1",
          },
        },
      ],
      "_type": "sentry_team",
      "createdOn": undefined,
      "displayName": "jupiterone-dev-testteam1",
      "id": "1254438",
      "name": "jupiterone-dev-testteam1",
      "slug": "jupiterone-dev-testteam1",
    },
    Object {
      "_class": Array [
        "UserGroup",
      ],
      "_key": "sentry-team-1252019",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "avatar": Object {
              "avatarType": "letter_avatar",
              "avatarUuid": null,
            },
            "dateCreated": "2021-10-26T20:55:38.704349Z",
            "externalTeams": Array [],
            "hasAccess": true,
            "id": "1252019",
            "isMember": false,
            "isPending": false,
            "memberCount": 2,
            "name": "JupiterOne Integration Development",
            "projects": Array [
              Object {
                "avatar": Object {
                  "avatarType": "letter_avatar",
                  "avatarUuid": null,
                },
                "color": "#b23fbf",
                "dateCreated": "2021-10-26T20:56:06.026232Z",
                "features": Array [
                  "alert-filters",
                  "issue-alerts-targeting",
                  "minidump",
                  "race-free-group-creation",
                  "similarity-indexing",
                  "similarity-view",
                ],
                "firstEvent": "2021-10-26T20:59:35.888000Z",
                "firstTransactionEvent": true,
                "hasAccess": true,
                "hasSessions": false,
                "id": "6036546",
                "isBookmarked": false,
                "isInternal": false,
                "isMember": false,
                "isPublic": false,
                "name": "jupiterone-integration-develop",
                "platform": "node",
                "slug": "jupiterone-integration-develop",
                "status": "active",
              },
              Object {
                "avatar": Object {
                  "avatarType": "letter_avatar",
                  "avatarUuid": null,
                },
                "color": "#3f5abf",
                "dateCreated": "2021-10-27T11:58:01.388720Z",
                "features": Array [
                  "alert-filters",
                  "issue-alerts-targeting",
                  "minidump",
                  "race-free-group-creation",
                  "similarity-indexing",
                  "similarity-view",
                ],
                "firstEvent": null,
                "firstTransactionEvent": false,
                "hasAccess": true,
                "hasSessions": false,
                "id": "6037645",
                "isBookmarked": false,
                "isInternal": false,
                "isMember": false,
                "isPublic": false,
                "name": "testpythonproject1",
                "platform": "python",
                "slug": "testpythonproject1",
                "status": "active",
              },
            ],
            "slug": "jupiterone-integration-develop",
          },
        },
      ],
      "_type": "sentry_team",
      "createdOn": undefined,
      "displayName": "JupiterOne Integration Development",
      "id": "1252019",
      "name": "JupiterOne Integration Development",
      "slug": "jupiterone-integration-develop",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "sentry-user-10763262",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "dateCreated": "2021-10-26T21:02:47.248596Z",
            "email": "adam.pierson@jupiterone.com",
            "expired": false,
            "flags": Object {
              "member-limit:restricted": false,
              "sso:invalid": false,
              "sso:linked": false,
            },
            "id": "10763262",
            "inviteStatus": "approved",
            "inviterName": "Austin Kelleher",
            "name": "Adam Pierson",
            "pending": false,
            "projects": Array [
              "jupiterone-integration-develop",
              "testproject2",
              "testpythonproject1",
            ],
            "role": "owner",
            "roleName": "Owner",
            "user": Object {
              "avatar": Object {
                "avatarType": "letter_avatar",
                "avatarUuid": null,
              },
              "avatarUrl": "https://secure.gravatar.com/avatar/5abd8230461366dd40ab803831f7ad97?s=32&d=mm",
              "dateJoined": "2021-10-26T21:08:09.627990Z",
              "email": "adam.pierson@jupiterone.com",
              "emails": Array [
                Object {
                  "email": "adam.pierson@jupiterone.com",
                  "id": "1735481",
                  "is_verified": true,
                },
              ],
              "experiments": Object {},
              "has2fa": true,
              "hasPasswordAuth": true,
              "id": "1601268",
              "isActive": true,
              "isManaged": false,
              "isStaff": false,
              "isSuperuser": false,
              "lastActive": "2021-11-09T19:13:38.496885Z",
              "lastLogin": "2021-11-02T17:49:11.974792Z",
              "name": "Adam Pierson",
              "username": "adam.pierson@jupiterone.com",
            },
          },
        },
      ],
      "_type": "sentry_member",
      "active": true,
      "createdOn": 1635282167248,
      "dateJoined": 1635282489627,
      "displayName": "Adam Pierson",
      "email": "adam.pierson@jupiterone.com",
      "id": "10763262",
      "isManaged": false,
      "isStaff": false,
      "isSuperuser": false,
      "lastActive": 1636485218496,
      "lastLogin": 1635875351974,
      "mfaEnabled": true,
      "name": "Adam Pierson",
      "role": "owner",
      "username": "adam.pierson@jupiterone.com",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "sentry-user-10763241",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "dateCreated": "2021-10-26T20:55:38.715181Z",
            "email": "austin.kelleher@jupiterone.dev",
            "expired": false,
            "flags": Object {
              "member-limit:restricted": false,
              "sso:invalid": false,
              "sso:linked": false,
            },
            "id": "10763241",
            "inviteStatus": "approved",
            "inviterName": null,
            "name": "Austin Kelleher",
            "pending": false,
            "projects": Array [
              "jupiterone-integration-develop",
              "testpythonproject1",
            ],
            "role": "owner",
            "roleName": "Owner",
            "user": Object {
              "avatar": Object {
                "avatarType": "letter_avatar",
                "avatarUuid": null,
              },
              "avatarUrl": "https://secure.gravatar.com/avatar/6c6d286122b52413957d2e2ed73deefc?s=32&d=mm",
              "dateJoined": "2021-10-26T20:55:38.564265Z",
              "email": "austin.kelleher@jupiterone.dev",
              "emails": Array [
                Object {
                  "email": "austin.kelleher@jupiterone.dev",
                  "id": "1735468",
                  "is_verified": true,
                },
              ],
              "experiments": Object {},
              "has2fa": false,
              "hasPasswordAuth": true,
              "id": "1601255",
              "isActive": true,
              "isManaged": false,
              "isStaff": false,
              "isSuperuser": false,
              "lastActive": "2021-11-09T18:18:43.196636Z",
              "lastLogin": "2021-11-08T19:34:22.952705Z",
              "name": "Austin Kelleher",
              "username": "austin.kelleher@jupiterone.dev",
            },
          },
        },
      ],
      "_type": "sentry_member",
      "active": true,
      "createdOn": 1635281738715,
      "dateJoined": 1635281738564,
      "displayName": "Austin Kelleher",
      "email": "austin.kelleher@jupiterone.dev",
      "id": "10763241",
      "isManaged": false,
      "isStaff": false,
      "isSuperuser": false,
      "lastActive": 1636481923196,
      "lastLogin": 1636400062952,
      "mfaEnabled": false,
      "name": "Austin Kelleher",
      "role": "owner",
      "username": "austin.kelleher@jupiterone.dev",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "sentry-organization-1052715",
      "_key": "sentry-organization-1052715|has|sentry-project-6036546",
      "_toEntityKey": "sentry-project-6036546",
      "_type": "sentry_organization_has_project",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "sentry-organization-1052715",
      "_key": "sentry-organization-1052715|has|sentry-project-6044646",
      "_toEntityKey": "sentry-project-6044646",
      "_type": "sentry_organization_has_project",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "sentry-organization-1052715",
      "_key": "sentry-organization-1052715|has|sentry-project-6037645",
      "_toEntityKey": "sentry-project-6037645",
      "_type": "sentry_organization_has_project",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "sentry-organization-1052715",
      "_key": "sentry-organization-1052715|has|sentry-team-1254438",
      "_toEntityKey": "sentry-team-1254438",
      "_type": "sentry_organization_has_team",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "sentry-organization-1052715",
      "_key": "sentry-organization-1052715|has|sentry-team-1252019",
      "_toEntityKey": "sentry-team-1252019",
      "_type": "sentry_organization_has_team",
      "displayName": "HAS",
    },
    Object {
      "_class": "ASSIGNED",
      "_fromEntityKey": "sentry-team-1254438",
      "_key": "sentry-team-1254438|assigned|sentry-project-6044646",
      "_toEntityKey": "sentry-project-6044646",
      "_type": "sentry_team_assigned_project",
      "displayName": "ASSIGNED",
    },
    Object {
      "_class": "ASSIGNED",
      "_fromEntityKey": "sentry-team-1252019",
      "_key": "sentry-team-1252019|assigned|sentry-project-6036546",
      "_toEntityKey": "sentry-project-6036546",
      "_type": "sentry_team_assigned_project",
      "displayName": "ASSIGNED",
    },
    Object {
      "_class": "ASSIGNED",
      "_fromEntityKey": "sentry-team-1252019",
      "_key": "sentry-team-1252019|assigned|sentry-project-6037645",
      "_toEntityKey": "sentry-project-6037645",
      "_type": "sentry_team_assigned_project",
      "displayName": "ASSIGNED",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "sentry-organization-1052715",
      "_key": "sentry-organization-1052715|has|sentry-user-10763262",
      "_toEntityKey": "sentry-user-10763262",
      "_type": "sentry_organization_has_member",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "sentry-organization-1052715",
      "_key": "sentry-organization-1052715|has|sentry-user-10763241",
      "_toEntityKey": "sentry-user-10763241",
      "_type": "sentry_organization_has_member",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "sentry-team-1254438",
      "_key": "sentry-team-1254438|has|sentry-user-10763262",
      "_toEntityKey": "sentry-user-10763262",
      "_type": "sentry_team_has_member",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "sentry-team-1252019",
      "_key": "sentry-team-1252019|has|sentry-user-10763262",
      "_toEntityKey": "sentry-user-10763262",
      "_type": "sentry_team_has_member",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "sentry-team-1252019",
      "_key": "sentry-team-1252019|has|sentry-user-10763241",
      "_toEntityKey": "sentry-user-10763241",
      "_type": "sentry_team_has_member",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "sentry_organization",
    "sentry_project",
    "sentry_organization_has_project",
    "sentry_team",
    "sentry_organization_has_team",
    "sentry_team_assigned_project",
    "sentry_member",
    "sentry_organization_has_member",
    "sentry_team_has_member",
  ],
  "numCollectedEntities": 8,
  "numCollectedRelationships": 13,
}
`;
